client/my-sites/domains/domain-management/dns/dns-add-new.jsx

```diff
@@ -137,23 +166,39 @@ class DnsAddNew extends React.Component {
 
 			const normalizedData = getNormalizedData(
 				formState.getAllFieldValues( this.state.fields ),
-				this.props.selectedDomainName
+				selectedDomainName
 			);
-			this.formStateController.resetFields( this.getFieldsForType( this.state.type ) );
-
-			this.props.addDns( this.props.selectedDomainName, normalizedData ).then(
-				() =>
-					this.props.successNotice( translate( 'The DNS record has been added.' ), {
-						duration: 5000,
-					} ),
-				( error ) =>
-					this.props.errorNotice(
-						error.message || translate( 'The DNS record has not been added.' )
-					)
+			if ( ! config.isEnabled( 'domains/dns-records-redesign' ) ) {
+				this.formStateController.resetFields( this.getFieldsForType( this.state.type ) );
+			}
+
+			if ( recordToEdit ) {
+				this.props.updateDns( selectedDomainName, [ normalizedData ], [ recordToEdit ] ).then(
+					() => this.handleSuccess( translate( 'The DNS record has been updated.' ) ),
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'The DNS record has been deleted.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/edit-dns-records)